git_greppers/update-single-repo.sh

```
@@ -0,0 +1,254 @@
+REPO_PATH=$(get_repo_path "$REPO_NAME")
+DRY_RUN=${2:-0}
+WITH_DIFF=${3:-"--with-diff"}
+WITH_CODE=${4:-"--include-code"}
+CHECKOUT_DEPTH=${CHECKOUT_DEPTH:-1}
+PREVIEW_MODE=${PREVIEW_MODE:-0}
+GENERATE_PR_PREVIEWS=${GENERATE_PR_PREVIEWS:-0}
+PR_DAYS_RANGE=${PR_DAYS_RANGE:-7}
+DEBUG_LEVEL=${DEBUG_LEVEL:-1}
+
+# Source utility functions and handlers
+source "$(pwd)/utils.sh" || { echo "Failed to source utils.sh"; exit 1; }
```

Shellcheck (0.10.0)

[info] Not following: ./utils.sh was not specified as input (see shellcheck -x).

(SC1091)